src/main/java/no/unit/bibs/contents/DBClient.java

```diff
@@ -76,15 +77,20 @@ private void initDynamoDbClient(Environment environment) {
      */
     public void createContents(ContentsDocument document) throws CommunicationException {
         try {
-            PutItemRequest putItemRequest = PutItemRequest
+            var putItemRequest = PutItemRequest
                     .builder()
                     .tableName(tableName)
                     .item(this.generateItemMap(document))
                     .build();
-            dbClient.putItem(putItemRequest);
-            logger.info("contents created");
+            var response = dbClient.putItem(putItemRequest);
+            if (isNull(response.sdkHttpResponse())) {
+                throw new BadGatewayException("No response from DynamoDB");
+            }
+            if (!response.sdkHttpResponse().isSuccessful()) {
+                throw new BadGatewayException("Failed to create ContentsDocument: " + response.sdkHttpResponse().statusCode());
```

Comment:
i wonder how the client application handles different error types and messages, but i think we will see that sooner or later ourselves